lib/rest-api.php

```diff
@@ -11,13 +11,22 @@
 }
 
 /**
- * Registers the Global Styles REST API routes.
+ * Overrides the REST controller for the `wp_global_styles` post type.
+ *
+ * @param array  $args      Array of arguments for registering a post type.
+ *                          See the register_post_type() function for accepted arguments.
+ * @param string $post_type Post type key.
+ *
+ * @return array Array of arguments for registering a post type.
  */
-function gutenberg_register_global_styles_endpoints() {
-	$global_styles_controller = new WP_REST_Global_Styles_Controller_Gutenberg();
-	$global_styles_controller->register_routes();
+function gutenberg_override_global_styles_endpoint( array $args, string $post_type ): array {
+	if ( 'wp_global_styles' === $post_type ) {
+		$args['rest_controller_class'] = WP_REST_Global_Styles_Controller_Gutenberg::class;
```

Comment:
Oh, that's what you [meant](https://github.com/WordPress/gutenberg/pull/65071#issuecomment-2345298670) 🤦🏻 

I just assumed we were doing this somewhere - but clearly not! 🤭 

Thanks!

Reply:
No override was working; it was just doing double work. Using a post-registration filter should be preferred when possible.

[Here](https://github.com/WordPress/gutenberg/pull/65071#issuecomment-2345260471), I meant why we were doing full override instead of extending the existing class and your [answer](https://github.com/WordPress/gutenberg/pull/65071#issuecomment-2345298670) was satisfying :)

Reply:
Oh, thanks for confirming. Good to know my powers of comprehension are not completely borked 🤣 

I smoked tested this PR only, but things are working as expected with no PHP errors. I'll give it a better test in the morning (AEST) if no-one gets to it first.

Reply:
Thank you, @ramonjd!

I see one PHP Unit test failure specific to PHP 8.2 env. Investigating that now.